api/ServerInput.cs

```diff
@@ -2,23 +2,77 @@
  * Input for reading device state from the Server API
  */
 using System;
+using System.Threading;
+using System.IO;
+using System.Diagnostics;
+using Newtonsoft.Json;
+using System.Net;
+using System.Threading.Tasks;
+using System.Text;
+using System.Net.Http;
 
 namespace api
 {
 public class ServerInput : IDeviceInput
 {
-	public ServerInput() 
+
+	string _serverURL;
+	public ServerInput(string server_URL)
+	{
+		_serverURL = server_URL;
+	}
+	public ServerInput()
 	{
+		_serverURL = "http://serverapi1.azurewebsites.net/";
 	}
 
-	/**
-	 * Updates device from the server
-	 */
 	public bool read(Device dev)
 	{
-		//TODO: Implement this function
-		return true;
+		//string json = JsonConvert.SerializeObject(dev);
+
+		try
+		{
+		var response = HttpRequest(dev);
+		if(response.StatusCode == HttpStatusCode.OK)
+		{
+			
+			bool update_success = Interfaces.UpdateDevice(dev, response.Content.ToString());
+			return true;
+		}
+		else
+		{
+			return false;
```

Comment:
There's a big simplification here - you can store `response.StatusCode == HttpStatusCode.Ok` in a variable, and simply return that value, instead of having the extra `else` and two return statements.